ops/terraform/services/server/modules/bfd_server_metrics/main.tf

```
@@ -7,18 +7,18 @@ locals {
   }
   namespace = "bfd-${local.env}/bfd-server"
   endpoints = {
-    all                = "*/fhir/*"
-    metadata           = "*/fhir/metadata*"
-    coverage_all       = "*/fhir/Coverage*"
-    patient_all        = "*/fhir/Patient*"
-    eob_all            = "*/fhir/ExplanationOfBenefit*"
-    claim_all          = "*/fhir/Claim"
-    claim_response_all = "*/fhir/ClaimResponse"
+    all                = ["*/fhir/*"]
+    metadata           = ["*/fhir/metadata*"]
+    coverage_all       = ["*/fhir/Coverage*"]
+    patient_all        = ["*/fhir/Patient*"]
+    eob_all            = ["*/fhir/ExplanationOfBenefit*"]
+    claim_all          = ["*/fhir/Claim", "*/fhir/Claim/"]
+    claim_response_all = ["*/fhir/ClaimResponse", "*/fhir/ClaimResponse/"]
```

Didnt you mention these endpoints have both trailing slash and non-trailing slash versions? Wouldnt we want to do 

```suggestion
    claim_all          = ["*/fhir/Claim", "*/fhir/Claim*"]
    claim_response_all = ["*/fhir/ClaimResponse", "*/fhir/ClaimResponse*"]
```
Similar to some of the others above to catch both?

---

Yep, there's two variations of each:

| Endpoint        | Non-slash Variant        | Slash Variant             |
t| --------------- | ------------------------ | ------------------------- |
| `Claim`         | `/v2/fhir/Claim`         | `/v2/fhir/Claim/`         |
| `ClaimResponse` | `/v2/fhir/ClaimResponse` | `/v2/fhir/ClaimResponse/` |

The `*` is similar to `.*` in regex, as [it captures one-or-more characters](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html#matching-terms-json-log-events) (a wildcard). So, if we were to specify `Claim` as `*/fhir/Claim*`, the same filter would _also_ capture `ClaimResponse` entries. This is why I opted for specifiying both filter variants explicitly.

We _could_ handle `ClaimResponse` similar to other endpoints (like `*/fhir/ClaimResponse*`), but doing some log filtering there aren't log entries for anything matching `ClaimResponse`/`Claim` where the URI ends with something else. So, explicitly specifying two filters for the non-trailing slash and trailing slash variant will catch all `ClaimResponse` entries. However, I'm OK with updating this _just_ for `ClaimResponse`, i.e.:

```suggestion
    claim_all          = ["*/fhir/Claim", "*/fhir/Claim/"]
    claim_response_all = ["*/fhir/ClaimResponse*"]
```

Hope that makes this a _little_ clearer.

---

In regex syntax `*` matches zero or more while `+` matches one or more.  Looking in the linked document it would seem that `e*` would match zero or more `e` characters.  Not an `e` followed by anything else.

> \*: Matches zero or more instances of the proceeding term. For example, %ab\*c% can match "ac", "abc", and "abbbc"; %ab[0-9]*% can match "ab", "ab0", and "ab129".

---

Sorry, the syntax and documentation is a bit confusing -- I wish I could link to a sentence rather than a particular section! 

That would apply if these filters were specified as `%*/fhir/ClaimResponse*%`, but since they're not wrapped with `%` characters the following filtering syntax rule applies for `*` (under the _Writing filter patterns for JSON log events_ sub-heading):

> You can enclose strings in double quotation marks (""). Strings that contain types other than alphanumeric characters and the underscore symbol must be placed in double quotation marks. **Use the asterisk ("*") as a wild card to match text.**

---

Thanks, forgot Claim* would match ClaimResponse, I knew there was something about that very thing back when that we fixed for these filters.